### ISSUE TITLE
os/user TestGroupIds failure "The user name could not be found" on Windows 10 AAD machine

### ISSUE THREAD

**Issue by dagood** [microsoft/go#108]
When I try to run the tests on Windows on my work machine, I get this error:

```
--- FAIL: TestGroupIds (0.00s)
    user_test.go:144: &{Uid:[...] Gid:[...] Username:NORTHAMERICA\dagood Name:Davis Goodin HomeDir:C:\Users\dagood}.GroupIds(): The user name could not be found.
FAIL
FAIL    os/user 1.819s
```

Someone else hit this in 2018, and it ended up seeming to be a configuration issue on their machine. Another machine also using AAD worked: https://github.com/golang/go/issues/26041. It doesn't look like they were able to figure out the underlying issue.

To check for AAD: https://stackoverflow.com/a/51852296

```
C:\Users\dagood>dsregcmd /status

+----------------------------------------------------------------------+
| Device State                                                         |
+----------------------------------------------------------------------+

             AzureAdJoined : YES
          EnterpriseJoined : NO
              DomainJoined : NO
               Device Name : dagood-3
[...]
```

**Comment by qmuntal** [comment]
I followed another approach to fix `TestGroupsIds` for AD-joined users: https://go-review.googlesource.com/c/go/+/611116. With that fix the test no longer fail on my machine. Can you give it a try @dagood? Thanks.

**Comment by dagood** [comment]
Yep, confirmed that `test os/user` fails on 1dfb33e8612d20f41cf4e034d9d0838abf75e04b (CL parent) with the same error as before and succeeds on 3ab54bb958465c6277aa3940a5645099550fd3c9 (CL).

(I re-ran the rest of `run.bat` locally for the first time in a while and it seems to have tons of other errors for me now too. 😢)
